### DISCUSSION TITLE
XML verification size does not match manifest size

### DISCUSSION THREAD

**Discussion by davidasheets** [Kalanyr/gogrepo#92] [Q&A] [answered]
I have used gogrepoc to download the vast majority of my GoG library (currently at just under 3TB) for offline archive. But for some reason, two games are causing me issues. In particular some files for pathfinder_wrath_of_the_righteous_base_game and
the_witcher_3_wild_hunt_game_of_the_year_edition_game

I get the following errors on only a few select files:

10:59:40 | preallocating '4504912' bytes for '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\setup_pathfinder_wrath_of_the_righteous_windows_2.0.7o.818_release_(62440).exe'
10:59:40 | preallocating '11164800' bytes for '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\patch_pathfinder_wrath_of_the_righteous_Windows_2.0.7n.815_Release_(62341)_to_Windows_2.0.7o.818_Release_(62440).exe'
10:59:41 | XML verification data size does not match manifest size for setup_pathfinder_wrath_of_the_righteous_windows_2.1.5r.935_release_(66442).exe. manifest 4504912, received 4565112, skipping.
10:59:41 | not moving uncompleted download '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\setup_pathfinder_wrath_of_the_righteous_windows_2.0.7o.818_release_(62440).exe', success: False remaining bytes: 4504912 / 4504912
10:59:41 | preallocating '188651136' bytes for '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\patch_pathfinder_wrath_of_the_righteous_Windows_2.0.7k.809_Release_(62132)_to_Windows_2.0.7n.815_Release_(62341).exe'
10:59:41 | XML verification data size does not match manifest size for setup_pathfinder_wrath_of_the_righteous_windows_2.1.5r.935_release_(66442)-1.bin. manifest 4290472446, received 4290412542, skipping.
10:59:41 | not moving uncompleted download '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\setup_pathfinder_wrath_of_the_righteous_windows_2.0.7o.818_release_(62440)-1.bin', success: False remaining bytes: 4290472446 / 4290472446
10:59:41 | preallocating '1177894752' bytes for '.\!downloading\pathfinder_wrath_of_the_righteous_base_game\patch_pathfinder_wrath_of_the_righteous_Windows_2.0.6l.789_Release_(61145)_to_Windows_2.0.7k.809_Release_(62132).exe'
10:59:41 | XML verification data size does not match manifest size for setup_pathfinder_wrath_of_the_righteous_windows_2.1.5r.935_release_(66442)-6.bin. manifest 4088087113, received 4294967294, skipping

I get errors like this on quite a few files, and it just leaves large but empty files in the !downloading directory. Am I doing something wrong, is there an issue with the manifest on GoG, or is there some bug? Thanks for any help.

**Comment by Kalanyr** [comment, marked as answer]
The usual cause of this is that the file has been updated in between when
you ran the update command and when you ran the download command. You
should force a refresh of the game in question using update with the  -ids
parameter and then try downloading again.

**Reply by davidasheets** [reply]
Thanks for the quick reply. Note that even when I run update (without the -ids parameter) and then download again, I get the same issue. So I must supply the specific -ids to force the update on these files?

**Comment by Kalanyr** [comment]
Yes, sometimes GOG does "stealth" updates where they change the files
without marking the game as updated so the standard update can't detect the
change ( it would require fetching information 1 layer deeper for every
game to detect this and it would make the routine update take about as long
as a full update)

**Reply by davidasheets** [reply]
Thanks so much! And really appreciate the script, it has helped me a lot.
